selfapi.js

```diff
@@ -407,97 +406,89 @@ API.prototype = {
         request: exampleRequest
       };
 
-      var request = client.request(requestOptions, function (response) {
-        var success = true;
+      beforeEachTest(function () {
```

Comment:
Actually, this is not the correct place to call this test setup function.

Because all examples are tested in parallel and in random order, `beforeEachTest` and `afterEachTest` should really be called just when the handler is actually about to get called. This implementation calls all setup functions one after the other, issues all test requests, but handlers are actually executed much later, causing some failures.

Maybe I should instrument the actual handler function to run a pre-handler and a post-handler. Or even better, if the tests have to run in parallel, maybe it's worth it to fork the node process as many times as there are examples to test, so that each test gets executed in its own environment independently from any other test.